checks/dcosrequirements/check.go

```diff
@@ -0,0 +1,307 @@
+/*
+This package implements checks for DC/OS system requirements, see e.g.,
+https://docs.d2iq.com/mesosphere/dcos/2.0/installing/production/system-requirements.
+*/
+
+package dcosrequirements
+
+import (
+	"bufio"
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/mesosphere/bun/v2/bundle"
+	"github.com/mesosphere/bun/v2/checks"
+)
+
+func init() {
+	builder := checks.CheckFuncBuilder{
+		CheckMasters:      collect,
+		CheckAgents:       collect,
+		CheckPublicAgents: collect,
+	}
+
+	check := checks.Check{
+		Name:           "dcos-requirements",
+		Description:    "Checks that nodes meet system requirements",
+		Cure:           "Adjust the node spec to meet system requirements.",
+		OKSummary:      "Nodes meet system requirements",
+		ProblemSummary: "Nodes do not meet system requirements",
+		Run:            builder.Build(),
+	}
+
+	checks.RegisterCheck(check)
+}
+
+func checkCpus(host bundle.Host) []error {
+	requirements := map[bundle.DirType]int{
```

Comment:
Just a minor comment, but I would recommend hoisting this and the other two requirement tables in the global scope on top, plus adding a link to the documentation where this is described.

My thinking is to reduce the space someone has to search if there is ever a change to these requirements in the future.